### PR TITLE
Index epoch data

### PR DESCRIPTION
**Pull request by obasilakis**
_[GitHub keywords to close any associated issues](https://blog.github.com/2013-05-14-closing-issues-via-pull-requests/)_

### Description

Updated the existing celo voter rewards table and fetcher (now celo epoch rewards). Now indexing all epoch data that the economics team is interested in. 

### Tested

Added tests for the changes

### Issues

 - Relates to #[issue number here]
 - Fixes #[issue number here]

 ### Backwards compatibility

 _Brief explanation of why these changes are/are not backwards compatible._

### Checklist

  - [X] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [ ] I added code comments for anything non trivial.
  - [ ] I added documentation for my changes.
  - [ ] If I added/changed/removed ENV var, I submitted a PR to https://github.com/celo-org/monorepo to update the list and default values of env vars.
  - [ ] If I add new indices into DB, I checked, that they are not redundant with PGHero or other tools.

### CODE REVIEW

File: apps/indexer/lib/indexer/fetcher/celo_epoch_rewards.ex [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        result = Chain.import(import_params)
+        {:ok, result}
+      end)
+      |> Multi.run(:delete_celo_pending, fn _, _ ->
+        success
+        |> Enum.each(fn reward -> Chain.delete_celo_pending_epoch_operation(reward.block_hash) end)
+
+        {:ok, success}
+      end)
+      |> Explorer.Repo.transaction()
```
Comment (rkachowski) [inline apps/indexer/lib/indexer/fetcher/celo_epoch_rewards.ex:104]: Normally the `Indexer` stays away from accessing the db directly, db transactions and import formatting is handled by the `Explorer` lib. See

* https://github.com/celo-org/blockscout/blob/57ded749ae41a71f1963cf186a0d5ffb1b1674f7/apps/indexer/lib/indexer/fetcher/internal_transaction.ex#L286
* https://github.com/celo-org/blockscout/blob/57ded749ae41a71f1963cf186a0d5ffb1b1674f7/apps/indexer/lib/indexer/block/catchup/fetcher.ex#L142
* https://github.com/celo-org/blockscout/blob/57ded749ae41a71f1963cf186a0d5ffb1b1674f7/apps/indexer/lib/indexer/fetcher/token.ex#L79

In these examples the importing of data to the db, and exact manner in which it's done is managed by a method in the `Explorer.Chain` context. This separates the concerns of fetching + processing data (`Indexer`), and verification + db import (`Explorer`).

tl;dr - this method needs to be moved into the `Explorer` app to avoid database logic + Ecto imports in the `Indexer`

Comment (obasilakis) [reply apps/indexer/lib/indexer/fetcher/celo_epoch_rewards.ex:104]: I thought about this and then saw that Ecto is indeed imported in some places in the `Indexer`, but you're right we shouldn't exacerbate this

**Review by rkachowski — APPROVED** [review]

File: apps/indexer/lib/indexer/fetcher/celo_epoch_rewards.ex [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  def fetch_from_blockchain(blocks) do
+    blocks
+    |> Enum.map(fn block ->
+      case AccountReader.validator_group_reward_data(block) do
+        {:ok, data} ->
+          data
+
+        error ->
+          Logger.debug(inspect(error))
```
Comment (carterqw2) [inline apps/indexer/lib/indexer/fetcher/celo_epoch_rewards.ex:57]: Is it a leftover? If not, then probably should have a different level.

File: apps/indexer/lib/indexer/fetcher/celo_epoch_rewards.ex [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            {[reward | failed], success}
+          end
+      end)
+
+    import_params = %{
+      celo_epoch_rewards: %{params: success},
+      timeout: :infinity
+    }
+
+    if failed != [] do
```
Comment (carterqw2) [inline apps/indexer/lib/indexer/fetcher/celo_epoch_rewards.ex:84]: What are the cases when it can fail?

File: apps/indexer/lib/indexer/fetcher/celo_epoch_rewards.ex [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        block_numbers: Enum.map(failed, fn rew -> rew.block_number end)
+      )
+    end
+
+    case Chain.import_epoch_rewards_and_delete_pending_celo_epoch_operations(import_params, success) do
+      {:ok, _} ->
+        :ok
+
+      {:error, reason} ->
+        Logger.debug(fn -> ["failed to import Celo voter reward data: ", inspect(reason)] end,
```
Comment (carterqw2) [inline apps/indexer/lib/indexer/fetcher/celo_epoch_rewards.ex:95]: Should be higher level as well?

File: apps/indexer/test/indexer/fetcher/celo_epoch_rewards_test.exs [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+
+      assert CeloEpochRewardsFetcher.init(
+               [],
+               fn block_number, acc -> [block_number | acc] end,
+               json_rpc_named_arguments
+             ) == [%{block_number: block.number, block_hash: block.hash}]
+    end
+
+    @tag :no_geth
+    test "does not buffer blocks with fetched internal transactions", %{
```
Comment (carterqw2) [inline apps/indexer/test/indexer/fetcher/celo_epoch_rewards_test.exs:54]: `fetched epoch rewards`?

**Review by carterqw2 — APPROVED** [review]
